R/backport_linter.R

```diff
@@ -19,36 +22,30 @@ backport_linter <- function(r_version = getRversion()) {
 
     xml <- source_expression$xml_parsed_content
 
-    names_xpath <- "//SYMBOL | //SYMBOL_FUNCTION_CALL"
     all_names_nodes <- xml2::xml_find_all(xml, names_xpath)
     all_names <- xml2::xml_text(all_names_nodes)
 
-    backport_blacklist <- backports[r_version < R_system_version(names(backports))]
-
-    # not sapply/vapply, which may over-simplify to vector -- cbind makes sure we have a matrix so rowSums works
-    needs_backport <- do.call(cbind, lapply(backport_blacklist, function(nm) all_names %in% nm))
-    bad_idx <- rowSums(needs_backport) > 0L
+    # not sapply/vapply, which may over-simplify to vector
+    # rbind makes sure we have a matrix with dimensions [n_versions x n_names]
+    # so that colSums() works to tell us which names are in an unavailable version
+    # rbind not cbind because R is column-major --> which() below will be in column order
+    needs_backport <- do.call(rbind, lapply(backport_blacklist, function(nm) all_names %in% nm))
```

Comment:
Can you compare performance to `vapply()` with a conditional `t()` if `length(n_names) == 1L`?
My experience is that `do.call(rbind, ...)` is extremely slow compared to `vapply()`

Reply:
```
needs_backport <- vapply(backport_blacklist, "%in%", logical(length(all_names)), x = all_names)
if (length(all_names) == 1L) needs_backport <- t(needs_backport)
```

Reply:
marking for follow-up so we can churn through these largely-rote PRs for #1199